src/cpp/session/modules/SessionConnections.R

```diff
@@ -517,4 +517,36 @@ options(connectionObserver = list(
    .rs.success()
 })
 
+.rs.addJsonRpcHandler("connection_test", function(code) {
+   error <- ""
 
+   oldConnectionObserver <- getOption("connectionObserver")
+   on.exit(options(connectionObserver = oldConnectionObserver))
+
+   disconnectCalls <- list()
+
+   options(connectionObserver = list(
+      connectionOpened = function(type, host, displayName, icon = NULL, 
+                                  connectCode, disconnect, listObjectTypes,
+                                  listObjects, listColumns, previewObject, 
+                                  connectionObject, actions = NULL) {
+         disconnectCalls <<- c(disconnectCalls, disconnect)
+      },
+      connectionClosed = function(type, host, ...) {
+
+      },
+      connectionUpdated = function(type, host, hint, ...) {
+      }
+   ))
+
+   .envir <- parent.frame(2)
```

Comment:
If the intention is to evaluate this within the 'active' frame, you might instead want to use `.rs.getActiveFrame()`, which should make sure this works in the (rare) case where e.g. the R debugger is active or similar.